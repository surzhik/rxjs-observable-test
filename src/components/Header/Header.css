/**
 * React Starter Kit (https://www.reactstarterkit.com/)
 *
 * Copyright Â© 2014-present Kriasoft, LLC. All rights reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.txt file in the root directory of this source tree.
 */

@import '../variables.css';

.headerHolder {
  width: 100%;
  margin-bottom: -40px;
  height: 40px;
}

.switchHolder {
  text-align: right;
  padding: 15px 30px 0 30px;
}
