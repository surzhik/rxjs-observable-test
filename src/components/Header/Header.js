/**
 * React Starter Kit (https://www.reactstarterkit.com/)
 *
 * Copyright Â© 2014-present Kriasoft, LLC. All rights reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.txt file in the root directory of this source tree.
 */

import React from 'react';
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import withStyles from 'isomorphic-style-loader/lib/withStyles';
import Switch from 'rc-switch';
import rc from 'rc-switch/assets/index.css';
import { setModeSwitch } from '../../actions/modeSwitch';
import s from './Header.css';

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators(
      {
        setModeSwitch,
      },
      dispatch,
    ),
  };
}

function mapStateToProps({ modeSwitch: { nightMode } }) {
  return {
    nightMode,
  };
}

export class Header extends React.PureComponent {
  /* eslint-disable react/forbid-prop-types */
  static propTypes = {
    nightMode: PropTypes.bool.isRequired,
    actions: PropTypes.object.isRequired,
  };

  handleSwitch = () => {
    const { actions, nightMode } = this.props;
    actions.setModeSwitch(!nightMode);
  };

  render() {
    const { nightMode } = this.props;
    return (
      <div className={s.headerHolder}>
        <div className={s.switchHolder}>
          <Switch checked={nightMode} onChange={this.handleSwitch} />
        </div>
      </div>
    );
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(withStyles(s, rc)(Header));
