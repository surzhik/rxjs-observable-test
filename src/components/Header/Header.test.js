/* eslint-env jest */
/* eslint-disable padded-blocks, no-unused-expressions */

import React from 'react';
import renderer from 'react-test-renderer';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Switch from 'rc-switch';
import { Header } from './Header';

configure({ adapter: new Adapter() });

describe('Header container', () => {
  const props = {
    nightMode: false,
    actions: {},
  };

  describe('Header structure', () => {
    test('renders children correctly', () => {
      const wrapper = renderer.create(<Header {...props} />).toJSON();

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('Header initial', () => {
    const headerContainer = shallow(<Header {...props} />);

    it('Render switch', () => {
      expect(headerContainer.find(Switch)).toHaveLength(1);
    });
  });
});
