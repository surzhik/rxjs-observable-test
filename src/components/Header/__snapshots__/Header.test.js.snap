// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header container Header structure renders children correctly 1`] = `
<div
  className="headerHolder"
>
  <div
    className="switchHolder"
  >
    <button
      aria-checked={false}
      checked={false}
      className="rc-switch"
      defaultChecked={false}
      onChange={[Function]}
      onClick={[Function]}
      onKeyDown={[Function]}
      onMouseUp={[Function]}
      role="switch"
      type="button"
    >
      <span
        className="rc-switch-inner"
      />
    </button>
  </div>
</div>
`;
